packages/package-requester/src/packageRequester.ts

```diff
@@ -111,7 +112,22 @@ export default function (
   const fetch = fetcher.bind(null, opts.fetchers, opts.cafs)
   const fetchPackageToStore = fetchToStore.bind(null, {
     // eslint-disable-next-line @typescript-eslint/no-unnecessary-boolean-literal-compare
-    checkFilesIntegrity: opts.verifyStoreIntegrity === false ? async () => true : _checkFilesIntegrity.bind(null, cafsDir),
+    checkFilesIntegrity: opts.verifyStoreIntegrity === false
+      ? async (pkgIndex, deferredManifest) => {
+        // If verifyStoreIntegrity is false we skip the integrity checks of all files
+        // and only read the package manifest.
+
+        const pkg = pkgIndex['package.json']
+
+        if (pkg && deferredManifest) {
```

Comment:
I guess if no pkg is found, deferredManifest should still be resolved.

Reply:
Resolve it with `undefined` then?